biomage/experiment/pull.py

```diff
@@ -36,6 +41,10 @@ def download_if_modified(bucket, key, filepath):
     s3 = boto3.resource("s3")
     s3_obj = s3.Object(bucket, key)
 
+    try:
+        s3_obj.last_modified
```

Comment:
The try statement seems odd -- is the idea to check if `s3_obj.last_modified` will result in a clienterror?